### ISSUE TITLE
Missing warning deprecated copy constructor warning in -Wextra

### ISSUE THREAD

**Issue by roe85** [llvm/llvm-project#95]
There seems to be some interference between  `-Wdeprecated-copy-dtor` and `-Wdeprecated-copy` which leads to emitting only the first one. 
This is bad because only the second is governed by `-Wextra`, which is likelier to be activated. The code attached is not warned about whith `-Wextra` unless the destructor is commented out. GCC warns in either case.
 I think there is no reason, why both warnings shouldn't be governed with `-Wextra`.

```
class A {
    public:
    void b();
    ~A();
    A& operator=(const A& a);

};
void test() {
    
    A a;

    A b = a;
    a = b;

    (void)a;
    (void)b;
}
```

**Comment by Quuxplusone** [comment]
This behavior was altered beyond recognition by the fix for #44979, so maybe this can be closed. `-Wdeprecated` now says:
```
<source>:6:5: warning: definition of implicit copy constructor for 'A' is deprecated because it has a user-provided destructor [-Wdeprecated-copy-with-user-provided-dtor]
    ~A();
    ^
<source>:14:11: note: in implicit copy constructor for 'A' first required here
    A b = a;
          ^
```
But this warning is still enabled only by `-Wdeprecated`, not by `-Wall` or `-Wextra`. I think it _would_ be nice to enable it under `-Wextra`, at least, since it does usually indicate a bug or at least a time bomb.
